src/lucky/routeable.cr

```diff
@@ -100,6 +118,18 @@ module Lucky::Routeable
 
       is_root_path = path == ""
       path = "/" if is_root_path
+
+      query_params = {} of String => String
+      {% for param in PARAM_DECLARATIONS %}
+        unless {{ param.var }} == {{ param.value || nil }}
+          query_params["{{ param.var }}"] = {{ param.var }}.to_s
+        end
+      {% end %}
+
+      unless query_params.empty?
+        path += "?#{HTTP::Params.encode(query_params)}"
+      end
```

Comment:
Just FYI, there is a lot of duplication, but I think that's because of how these methods are written. I'm going to take a look at these later to see how they can be cleaned up. I think I might remove `path` and instead just have `with/route`. Then if you want the path you could do `User::Show.with(user.id).path` I'm not sure though